core/src/mindustry/logic/LExecutor.java

```diff
@@ -580,6 +601,16 @@ public void run(LExecutor exec){
 
             if(from instanceof MemoryBuild mem && (exec.privileged || (from.team == exec.team && !mem.block.privileged)) && address >= 0 && address < mem.memory.length){
                 mem.memory[address] = value.num();
+            }else if(from instanceof LogicBuild logic && (exec.privileged || from.team == exec.team) && position.isobj && position.objval instanceof String name){
+                //limit write, need target processor set writable.
+                LVar writable = logic.executor.optionalVar("@writable");
```

Comment:
This is very strange. It's completely un-intuitive, and I don't understand why this variable has to be set.

Reply:
memory protection - you wouldn't want to break world processor maps by accident; and sometimes you want to use world processors for API or something

really the statement below should also check for privilege

Reply:
See other comment - this was already patched to check for privileged access.